client/core/core.go

```diff
@@ -3112,6 +3112,21 @@ func (c *Core) setWalletPassword(wallet *xcWallet, newPW []byte, crypter encrypt
 		}
 		wallet.setEncPW(encNewPW)
 	} else {
+		// Test that the wallet is actually good with no password. At present,
+		// this means the backend either cannot be locked or unlocks with an
+		// empty password. This might change. We will ignore several errors in
+		// this process since if the external wallet's password was removed
+		// without a reconnect, the backend state may be inconsistent without a
+		// full reconnect with a fresh wallet.
+		backend := wallet.Wallet
+		_ = backend.Lock()           // may error, but will not actually lock for unprotected wallet
+		_ = backend.Unlock([]byte{}) // some backends may error, others require to avoid a reconnect
```

Comment:
> others require to avoid a reconnect

Is this referring to Electrum?

Reply:
I suppose it would be any wallet that lets you go from protected->unprotected (or the reverse) while the wallet is running _and DEX is already connected to it_.  Off hand, I'm not sure if Bitcoin Core lets you remove the passphrase or not.  But yes, this currently addresses the possibility with Electrum wallets how I can remove password protection and our code won't find out until reconnect unless we do some prodding here at time of reconfigure.

![image](https://user-images.githubusercontent.com/9373513/193641688-ee32233c-8e2b-4ee6-9c75-6b79e0f5909b.png)

For reference, we discover this on wallet connect for Bitcoin Core by inspecting a `getwalletinfo` response field, while for Electrum we test with a getprivatekey RPC to see if it works with no pass.  AFAICT we had a bit of a hole in the reconfigure logic that would always let you remove the password even if that was not correct for the wallet.